lib/one_model.ex

```diff
@@ -393,12 +409,20 @@ defmodule OneModel do
       @doc """
       Delete the #{@schema} given a changeset, or raise an exception.
       """
-      def delete!(%Ecto.Changeset{} = changeset), do: @repo.delete!(changeset)
+      def delete!(%Ecto.Changeset{} = changeset) do
+        case delete(changeset) do
+          {:ok, schema} ->
+            schema
+
+          {:error, changeset} ->
+            raise Ecto.InvalidChangesetError, changeset: changeset, action: :delete
+        end
+      end
 
       @doc """
       Delete the given #{@schema} by id, or raise an exception.
       """
-      def delete!(id), do: id |> get() |> delete!()
+      def delete!(id), do: id |> get() |> change() |> delete!()
```

Comment:
`delete/1` accepts a changeset, schema, or primary id. `delete!/1` neglects  one of those, which I don't think is correct. I think there should only be one function clause which simply does a pattern match on the return of `delete/1` without any additional logic. Just:
```
      @spec delete!(Ecto.Changeset.t | @schema.t |  Ecto.UUID.t) do
      def delete!(entry) do
        case delete(entry) do
          {:ok, schema} ->
            schema

          {:error, changeset} ->
            raise Ecto.InvalidChangesetError, changeset: changeset, action: :delete
        end
      end
```
This is analogous to `create!` which is also only /1.

Reply:
delete!/1 does not ignore anything. Please see lines 407, 412 and 425.

However your point about consolidating those into one is good.